src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java

```diff
@@ -553,4 +553,46 @@ public void setNewGameMode(GameType newGameMode)
             this.newGameMode = newGameMode;
         }
     }
+
+    /**
+     *
+     * Fired when a player clicks the "done" button in the sign edit UI.
+     * Cancelling will set the edited sign to be blank (i.e. lines will be an empty list)
+     *
+     */
+    @Cancelable
+    public static class SignChangeEvent extends PlayerEvent
+    {
+
+        private final BlockPos pos;
+        private List<String> lines;
+
+        public SignChangeEvent(Player player, BlockPos pos, List<String> lines)
+        {
+            super(player);
```

Comment:
Since this event is fired on the (logical) server, it may be good to override `getPlayer()` and have it return a `ServerPlayer` (thanks to covariant return types in Java).